pybamm/solvers/casadi_solver.py

```diff
@@ -215,6 +244,35 @@ def _integrate(self, model, t_eval, inputs=None):
                         ]
                     )
                 )
+
+                extrap_event = [
+                    event(t, current_step_sol.y[:, -1], inputs=inputs)
+                    for event in model.interpolant_extrapolation_events_eval
+                ]
+
+                if extrap_event:
+                    if (np.concatenate(extrap_event) < self.extrap_tol).any():
+                        extrap_event_names = []
+                        for event in model.events:
+                            if (
+                                event.event_type
+                                == pybamm.EventType.INTERPOLANT_EXTRAPOLATION
+                                and (
+                                    event.expression.evaluate(
+                                        t,
+                                        current_step_sol.y[:, -1].full(),
+                                        inputs=inputs,
+                                    )
+                                    < self.extrap_tol
+                                ).any()
+                            ):
+                                extrap_event_names.append(event.name[12:])
+
+                        raise pybamm.SolverError(
+                            "CasADI solver failed because the following interpolation"
+                            " bounds were exceeded: {}".format(extrap_event_names)
```

Comment:
Maybe indicate here how this can be fixed? e.g. "You may need to provide additional interpolation points outside these bounds"

Reply:
Done